index.js

```diff
@@ -308,4 +310,30 @@ module.exports = function configParser({
                 errors: [err.toString()]
             }))
     );
+}
+
+/**
+ * Parses pipeline template configuration
+ * @method parsePipelineTemplate
+ * @param   {Object}          config
+ * @param   {String}          config.yaml           Pipeline Template
+ * @return  {Object}                                Pipeline Template
+ */
+async function parsePipelineTemplate({ yaml }) {
+    const configToValidate = await parseYaml(yaml);
+    const pipelineTemplate = await SCHEMA_PIPELINE_TEMPLATE.validateAsync(configToValidate, {
+        abortEarly: false
+    });
+    const pipelineTemplateConfig = pipelineTemplate.config;
+
+    // flatten pipeline template shared setting into jobs
+    pipelineTemplateConfig.jobs = flattenSharedIntoJobs(pipelineTemplateConfig.shared, pipelineTemplateConfig.jobs);
+    delete pipelineTemplateConfig.shared;
+
+    return pipelineTemplate;
+}
```

Comment:
where would this function be used? does this function need to be incorporated as well into `screwdriver-models` package?

Reply:
validator will call this, then screwdriver api calls validator https://github.com/screwdriver-cd/screwdriver/blob/11813f8ac54a3b3bed4e4298ce7ba754c7b7d992/plugins/pipelines/templates/create.js#L22